samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,512 @@
+                                        symbolSwatches[childSourceSettings.ruleBehavior]?.let { swatch ->
+                                            add(SnapSourceProperty(childSource.name, swatch, childSourceSettings))
+                                        }
+
+                                        // Set the appropriate symbol for the sublayer based on the SnapRuleBehavior.
+                                        childSource.renderer =
+                                            SimpleRenderer(symbols[childSourceSettings.ruleBehavior])
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Clears the selection on the layer and reinstates feature visibility, then resets the selected
+     * feature and layer and any UI backing variables.
+     */
+    private fun resetSelections() {
+        // Clear the existing selection and show the selected feature                                      
+        (selectedFeature?.featureTable?.layer as? FeatureLayer)?.clearSelection()                          
+        (selectedFeature?.featureTable?.layer as? FeatureLayer)?.setFeatureVisible(selectedFeature!!, true)
+
+        // Reset the selected feature and layer
+        selectedFeature = null
+        _assetGroupNameState.value = "<Nothing selected>"
+        _assetTypeNameState.value = "<Nothing selected>"
+        _isEditButtonEnabled.value = false
+
+        // Revert back to the default renderer for the distribution and service pipe layers and
+        // graphics overlay
+        distributionPipeLayer?.renderer = defaultDistributionRenderer
+        servicePipeLayer?.renderer = defaultServiceRenderer
+        graphicsOverlay.renderer = defaultGraphicRenderer
+
+        // Clear the snap sources list
+        _snapSourcePropertyList.value = emptyList()
+    }
+
+    /**
+     * Adds required layers from the geodatabase to the map, setting the visibility of sublayers to
+     * show only a small subset in order to avoid too much visual clutter.
+     */
+    private suspend fun addLayersToMapFromGeodatabase(application: Application) {
+        // Only show the Distribution Pipe and Service Pipe sublayers in the pipeline layer. Also
+        // store the default renderer for the these sublayers.
+        val pipeLayer = SubtypeFeatureLayer(
+            geodatabase.getFeatureTable(pipelineLayerName) as ArcGISFeatureTable
+        )
+        pipeLayer.load().getOrElse {
+            messageDialogVM.showMessageDialog(
+                "Error loading pipeline layer",
+                it.message.toString()
+            )
+        }
+        val distributionPipeName = application.getString(R.string.distribution_pipe_name)
+        val servicePipeLayerName = application.getString(R.string.service_pipe_layer_name)
+        pipeLayer.subtypeSublayers.forEach { sublayer ->
```

Comment:
```suggestion
        val servicePipeLayerName = application.getString(R.string.service_pipe_layer_name)
        // Assign the correct sublayer and renderer to each subtype sublayer
        pipeLayer.subtypeSublayers.forEach { sublayer ->
```

Or another comment that makes sense here

Reply:
Used comment text from reference .NET implementation.